src/librustc/ich/fingerprint.rs

```diff
@@ -46,6 +49,21 @@ impl Fingerprint {
         format!("{:x}{:x}", self.0, self.1)
     }
 
+    pub fn encode_opaque(&self, encoder: &mut Encoder) -> EncodeResult {
+        let bytes: [u8; 16] = unsafe { mem::transmute((self.0, self.1)) };
+
+        encoder.emit_raw_bytes(&bytes)
+    }
+
+    pub fn decode_opaque<'a>(decoder: &mut Decoder<'a>) -> Result<Fingerprint, String> {
+        let mut bytes = [0; 16];
+
+        decoder.read_raw_bytes(&mut bytes)?;
+
+        let (l, r): (u64, u64) = unsafe { mem::transmute(bytes) };
+
+        Ok(Fingerprint(l, r))
```

Comment:
... the following here:
```rust
Ok(Fingerprint(u64::from_le(l), u64::from_le(r)))
```
(If you get a method resolution error, you might have to `use std::u64;`)